### PR TITLE
relations header button to appear at all times for mobile devices

### LINKED ISSUE #7036: Always display icons to add record, not just on hover
## Scope & Context

- When a new user creates an account with Twenty and only operates it though mobile, then they can't get to know that sections like Account Owner, People and Opportunities have dropdowns. On large devices, there is hover so they know the actions and features.

https://github.com/user-attachments/assets/09ce7588-a960-4f7a-baa5-a04fdbee528e

- on mobile viewport

https://github.com/user-attachments/assets/36b5b5d2-298d-4195-b5df-10a2dd5d7009

- For views, on light tap edit button does show up but no clue until not tapped.
Screenshot 2024-09-15 at 6 08 44 PM

## Expected behavior

- We can introduce tooltips or information badges for first time user or we can get to a better desing approach.
- On mobile viewports, user has no idea that there exists dropdowns or actions on blank space until clicked on them
- We could use  `:focus ` which is closest to hover for mobile viewports or animations.

### PR DESCRIPTION
**Pull request by ehconitin**
fixes #7036

### CODE REVIEW

File: packages/twenty-front/src/modules/object-record/record-show/record-detail-section/components/RecordDetailRelationSection.tsx [hunk 1]
```diff
@@ -54,6 +55,7 @@ export const RecordDetailRelationSection = ({
   } = fieldDefinition.metadata as FieldRelationMetadata;
   const record = useRecoilValue(recordStoreFamilyState(recordId));
 
+  const isMobile = useIsMobile();
```
Comment (ehconitin) [inline packages/twenty-front/src/modules/object-record/record-show/record-detail-section/components/RecordDetailRelationSection.tsx:58]: Or should I pass in the prop in either Context or to the component from RecordShowContainer?

File: packages/twenty-front/src/modules/object-record/record-show/record-detail-section/components/RecordDetailRelationSection.tsx [hunk 2]
```diff
@@ -177,7 +179,7 @@ export const RecordDetailRelationSection = ({
               onClose={handleCloseRelationPickerDropdown}
               clickableComponent={
                 <LightIconButton
-                  className="displayOnHover"
+                  className={isMobile ? '' : 'displayOnHover'}
```
Comment (bosiraphael) [inline packages/twenty-front/src/modules/object-record/record-show/record-detail-section/components/RecordDetailRelationSection.tsx:182]: Is this line needed?

**Review by bosiraphael — COMMENTED** [review]
Hello @ehconitin ! Thank you for contributing :) LGTM, left a comment

File: packages/twenty-front/src/modules/object-record/record-show/record-detail-section/components/RecordDetailRelationSection.tsx (hunk 2 above)
Comment (ehconitin) [reply packages/twenty-front/src/modules/object-record/record-show/record-detail-section/components/RecordDetailRelationSection.tsx:182]: It doesn’t really make a difference; I just added it for code clarity.

Comment (bosiraphael) [reply packages/twenty-front/src/modules/object-record/record-show/record-detail-section/components/RecordDetailRelationSection.tsx:182]: Okay, I think we can leave it as it was then :)

Comment (ehconitin) [reply packages/twenty-front/src/modules/object-record/record-show/record-detail-section/components/RecordDetailRelationSection.tsx]: Done!

**Review by bosiraphael — APPROVED** [review]
LGTM

**Pull request merged**